### PR TITLE
tools/ci: enhance print_toolchain script

### PR DESCRIPTION
**Pull request by smlng**
### Contribution description

this PR enhances the tool script to print versions of installed compilers, devel libs and tools:

    - add generic command version printer
    - add cmake version
    - enhance output structure

### Issues/PRs references

None

### CODE REVIEW

File: dist/tools/ci/print_toolchain_versions.sh [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-    local ver=
-    if command -v "$cc" 2>&1 >/dev/null; then
-        ver=$("$cc" --version | head -n 1)
-    fi
-    if [ -z "$ver" ]; then
-        ver=missing/error
+
+    local cmd="$1"
+    if command -v "$cmd" 2>&1 >/dev/null; then
+        ver=$("$cmd" --version | head -n 1)
```
Comment (gebart) [inline dist/tools/ci/print_toolchain_versions.sh]: Please add
```sh
if [ -z "$ver" ]; then
    ver = error
fi
```
to detect some other errors

**Comment by smlng** [comment]
> newlib_version and avr_libc_version can be simplified in the same way 

not quite, at least how they use the `get_define` function differs.

**Comment by gebart** [comment]
@smlng I meant change the argument to avr_libc_version into "avr-gcc" and newlib_version argument to "${p}-gcc"

**Comment by smlng** [comment]
ah, now I get it ... yeah will do

File: dist/tools/ci/print_toolchain_versions.sh [hunk 2]
```diff
@@ -24,83 +25,52 @@ get_define() {
...
     printf "%s" "$line"
 }
 
 newlib_version() {
     if [ -z "$1" ]; then
-        cc=gcc
-    else
-        cc=$1-gcc
+        printf "%s" "error"
     fi
```
Comment (gebart) [inline dist/tools/ci/print_toolchain_versions.sh]: Missing return above

File: dist/tools/ci/print_toolchain_versions.sh [hunk 3]
```diff
@@ -24,83 +25,52 @@ get_define() {
...
-}
-
-git_version() {
-    local cmd="git"
-    if command -v "$cmd" 2>&1 >/dev/null; then
-        ver=$("$cmd" --version | head -n 1)
-    else
-        ver="missing"
+        printf "%s" "error"
     fi
```
Comment (gebart) [inline dist/tools/ci/print_toolchain_versions.sh]: Missing return above

**Comment by smlng** [comment]
@gebart can you have another look, I amended changes as suggested and squashed.

**Pull request merged**